cmd/spiffe-aws-assume-role/cli/iso8601DurationMapper.go

```diff
@@ -0,0 +1,30 @@
+package cli
+
+import (
+	"github.com/alecthomas/kong"
+	iso8601duration "github.com/channelmeter/iso8601duration"
+	"reflect"
+)
+
+const iso8601DurationMapperType = "iso8601duration"
+
+type iso8601DurationMapper struct {
+	kong.Mapper
+}
+
+func (iso8601DurationMapper) Decode(context *kong.DecodeContext, target reflect.Value) error {
+	token, err := context.Scan.PopValue(iso8601DurationMapperType)
+	if err != nil {
+		return err
+	}
+
+	rawValue := token.Value.(string)
+	isoDuration, err := iso8601duration.FromString(rawValue)
+	if err != nil {
+		return err
+	}
+	duration := isoDuration.ToDuration()
+	target.Set(reflect.ValueOf(duration))
+
+	return nil
+}
```

Comment:
This is our logic for taking in an ISO 8601 duration string and returning (well, via pointers) a time.Duration.